sqlglot/dialects/dialect.py

```diff
@@ -498,13 +499,14 @@ def _if_sql(self: Generator, expression: exp.If) -> str:
     return _if_sql
 
 
-def arrow_json_extract_sql(self: Generator, expression: exp.JSONExtract | exp.JSONBExtract) -> str:
-    return self.binary(expression, "->")
+def arrow_json_extract_sql(self: Generator, expression: exp.JSONExtract) -> str:
+    this = expression.this
+    if self.JSON_TYPE_REQUIRED_FOR_EXTRACTION and isinstance(this, exp.Literal) and this.is_string:
+        this.replace(exp.cast(this.copy(), "json"))
 
+    if type(expression) in (exp.JSONExtract, exp.JSONBExtract):
```

Comment:
maaybe this should be isinstance, and jsonextracscalar shouldn't be a subclass?

Reply:
yeah I was thinking about this, not sure why these are subclasses of `JSONExtract`

Reply:
shouldn't this only be true if size of json path is one?

Reply:
how do we handle multiple paths?

Reply:
`JSONExtract` by definition has one path, this is supposed to be a helper to generate arrows. Multiple paths would be parsed into nested `JSONExtract` expressions and each of those would be generated separately.

Reply:
what i mean is a json path with nested calls,

json_path(x, '$.a.b.c')

Reply:
i see what you mean, but just to make sure: your concern is what happens when we generate sql for dialects that don't support the json path syntax and we need to split these paths into multiple extractions?

Reply:
well for postgres, it would be json_extract_path(x, 'a', 'b', 'c')